src/tribler/core/components/tunnel/community/tunnel_community.py

```diff
@@ -503,7 +504,7 @@ def monitor_downloads(self, dslist):
             if state_changed and new_state in active:
                 if old_state != DownloadStatus.METADATA or new_state != DownloadStatus.DOWNLOADING:
                     self.join_swarm(info_hash, hops[info_hash], seeding=new_state == DownloadStatus.SEEDING,
-                                    callback=lambda addr, ih=info_hash: self.on_e2e_finished(addr, ih))
+                                    callback=functools.partial(self.on_e2e_finished, info_hash=info_hash))
```

Comment:
The new version appears to be more complex. Please contemplate retaining the lambda, or consider substituting it with a function, such as:

```python
                    def callback(addr, ih=info_hash):
                        return self.on_e2e_finished(addr, ih)

                    self.join_swarm(info_hash, hops[info_hash], seeding=new_state == DownloadStatus.SEEDING,
                                    callback=callback)
```

Reply:
Can you please explain why it appears more complex? It does the same thing. And afacit, it has the same information😅.

Reply:
I surmise that a larger number of code readers are accustomed to lambdas rather than `functools.partial`. Hence, they might find the lambda version more readable. Since there are no significant advantages of using `functools.partial` over lambdas in this particular context, I believe the lambda version is preferable. Does that make sense?

Reply:
Yes. I'll reverse the change.